contracts/MoMABurn.sol

```diff
@@ -0,0 +1,136 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+import "./FeralfileArtworkV3_3.sol";
+
+contract MoMABurn is Ownable {
+    using SafeMath for uint256;
+    address constant DEAD_ADDRESS = 0x000000000000000000000000000000000000dEaD;
+
+    ERC721 public burnContract; // Address of token need to burn
+    FeralfileExhibitionV3_3 public tier1Contract; // Address of new token for Tier 1
+    FeralfileExhibitionV3_3 public tier2Contract; // Address of new token for Tier 2
+    FeralfileExhibitionV3_3 public tier3Contract; // Address of new token for Tier 3
+
+    uint256 private tier1ArtworkID; // Artwork ID of Tier 1
+    uint256 private tier2ArtworkID; // Artwork ID of Tier 2
+    uint256 private tier3ArtworkID; // Artwork ID of Tier 3
+
+    bool public isBurnEnabled;
+
+    constructor() {
+        isBurnEnabled = false;
+    }
+
+    /// @notice setBurnContract use for update burn contract address
+    /// @param _contract - new burn contract address
+    function setBurnContract(address _contract) public onlyOwner {
+        require(_contract != address(0), "Invalid contract address");
+        burnContract = ERC721(_contract);
+    }
+
+    /// @notice setMintContracts use for update mint contract addresses
+    /// @param _tier1Address - new burn contract address for tier 1
+    /// @param _tier2Address - new burn contract address for tier 2
+    /// @param _tier3Address - new burn contract address for tier 3
+    function setMintContracts(
+        address _tier1Address,
+        address _tier2Address,
+        address _tier3Address
+    ) public onlyOwner {
+        require(
+            _tier1Address != address(0) &&
+                _tier2Address != address(0) &&
+                _tier3Address != address(0),
+            "Invalid contract address"
+        );
+        tier1Contract = FeralfileExhibitionV3_3(_tier1Address);
```

Comment:
somehow, I feel we can only save addresses. And only make the instances when we call to it.

Reply:
Updated